### PR TITLE
fixed redistribution between GOCART dust/sea salt and MOZAIC bins

### PR DESCRIPTION
**Pull request by saneku**
Bug fix in redistribution between GOCART dust/sea salt and MOZAIC bins in optical properties.

TYPE: bug fix

KEYWORDS: GOCART, optical properties, dust, sea salt, AOD

SOURCE: KAUST

DESCRIPTION OF CHANGES:
Problem:
It was found that WRF-Chem with GOCART aerosol scheme disregards the radiative effect of the largest dust and sea salt particles (5th and 4th bins).

Solution:
Mass redistribution between GOCART dust/sea salt and MOZAIC bins now accounts for the 5th dust and 4th sea salt bins. The number of MOZAIC bins was changed from 8 to 9 and MOZAIC max size range was increased from 10 to 20 um.

Table: Radii ranges (um) of dust and sea salt bins used in the GOCART aerosol module.
Screenshot 2024-09-25 at 10 20 26

Table: Size  (um) ranges for MOSAIC grid used in calculation of optical properties. Last row - added 9th bin.
MOS1: 0.039-0.078, 
MOS2: 0.078-0.156, 
MOS3: 0.156-0.312, 
MOS4: 0.312-0.625, 
MOS5: 0.625-1.25, 
MOS6: 1.25-2.5, 
MOS7: 2.5-5.0, 
MOS8: 5.0-10.0,
**MOS9: 10.0-20.0**

Table: Mass redistribution between GOCART dust/sea salt and MOZAIC bins. Numbers in bold - after modification.
Screenshot 2024-09-25 at 10 19 46

This PR is an addition to  #667 and #677 PR.

LIST OF MODIFIED FILES: 
M	chem/module_optical_averaging.F
M	chem/optical_driver.F

TESTS CONDUCTED: 
7-day WRF-Chem run over the Middle East with the GOCART aerosol scheme was conducted without the changes (BEFORE) and with changes (AFTER). The figure below compares time-averaged AOD fields for both runs. As expected, AOD increased by 3% on average. Over dust source regions the AOD increase is more pronounced - 5% on average.
aod_aver

RELEASE NOTE: Bug fix in the calculation of optical properties. Mass redistribution between GOCART dust/sea salt and MOZAIC bins was corrected. It slightly increased (by 3-5%) the aerosol optical depth (AOD).

### CODE REVIEW

**Comment by weiwangncar** [comment]
The regression test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: chem/module_optical_averaging.F [hunk 1]
```diff
@@ -3692,47 +3695,51 @@ subroutine optical_prep_gocart(nbin_o, chem, alt,relhum,          &
...
+        WRITE(*,*)'Dust redistribution:'
+        do m =1, ndust
+            WRITE(*,*)m,dustfrc_goc9bin(m,:)
+        end do
+
+        WRITE(*,*)'Sea salt redistribution:'
+        do m =1, 4
+            WRITE(*,*)m,seasfrc_goc9bin(m,:)
+        end do
+        !-----
```
Comment (weiwangncar) [inline chem/module_optical_averaging.F:3742]: Unless this is controlled by a switch, these write statements should be removed or commented out for general use of the code.

**Comment by weiwangncar** [comment]
@jordanschnell Just wondering if you could help review this PR - if it is not too involved.... Thanks!

**Review by jordanschnell — DISMISSED** [review]
The changes look good to me. Approved.

**Review by jordanschnell — APPROVED** [review]
Approved

**Comment by weiwangncar** [comment]
@jordanschnell Greatly appreciated!

**Review by weiwangncar — APPROVED** [review]

**Comment by saneku** [comment]
@islas Could you please add this bug fix description to the v4.6.1 release in the chemistry section?

**Comment by islas** [comment]
@saneku It looks like this has gone into the develop branch. This would be included in the v4.7.0 release

Sorry for the confusion, especially in the midst of the v4.6.1 bug fix release being made before this PR is published in a release.

**Comment by weiwangncar** [comment]
@saneku Sorry that I didn't check the branch your PR was based before merging...

**Pull request merged**